### PR TITLE
Connecting front backend login and signup page - stores auth token and username in datastore

### PR DESCRIPTION
**Pull request by Puvikaran2001**
Connected login and signup page to backend.
Made an account page where the user can logout of the app.
Store the auth token and username in a DataStore, meaning stores the auth token and username, such that when reopening the app, it will show the home page if the user is logged in before closing the app.

IAccountRepository: Interface used to separate the ui layer and the data layer
AccountDataSource: Provides function to store and retrieve username and auth token from disk

### CODE REVIEW

**Review by S3JER — CHANGES_REQUESTED** [review]

**Review by Darkros1245 — APPROVED** [review]

File: frontend/app/src/main/java/dk/scheduling/schedulingfrontend/datasources/AccountDataSource.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                emit(emptyPreferences())
+            } else {
+                throw exception
+            }
+        }
+    }
+
+    private fun getValueFromKey(key: Preferences.Key<String>): Flow<String> {
+        return getData().map { account ->
+            account[key] ?: throw UserNotLoggedIn("The user is not logged in!")
```
Comment (mads256h) [inline frontend/app/src/main/java/dk/scheduling/schedulingfrontend/datasources/AccountDataSource.kt:43]: Rename UserNotLoggedIn To UserNotLoggedInException

File: frontend/app/src/main/java/dk/scheduling/schedulingfrontend/datasources/AccountDataSource.kt [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        accountDataStore.edit { account ->
+            account[key] = value
+        }
+    }
+
+    suspend fun getAuthToken(): UUID {
+        try {
+            val stream = getValueFromKey(AccountDataStoreKeys.AUTH_TOKEN_KEY)
+            return UUID.fromString(stream.first())
+        } catch (e: Throwable) {
```
Comment (mads256h) [inline frontend/app/src/main/java/dk/scheduling/schedulingfrontend/datasources/AccountDataSource.kt:60]: This seems a bit overkill to catch all throwables.

Comment (mads256h) [reply frontend/app/src/main/java/dk/scheduling/schedulingfrontend/datasources/AccountDataSource.kt:60]: At least pass the exception to UserNotLoggedInException and save it in an inner exception field so we can see what went wrong.

File: frontend/app/src/main/java/testdata/DummyAccountRepository.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package testdata
+
+import dk.scheduling.schedulingfrontend.repositories.account.IAccountRepository
+import java.util.UUID
+
+class DummyAccountRepository : IAccountRepository {
+    override suspend fun getAuthToken(): UUID {
+        return UUID.randomUUID()
```
Comment (mads256h) [inline frontend/app/src/main/java/testdata/DummyAccountRepository.kt:8]: Maybe not return a new random UUID every time we get the auth token. Seems pretty unexpected.

**Review by mads256h — CHANGES_REQUESTED** [review]
Small stuff

**Review by S3JER — APPROVED** [review]

**Pull request merged**